cou/apps/base.py

```diff
@@ -713,3 +717,43 @@ def _get_wait_step(self) -> PostUpgradeStep:
             parallel=False,
             coro=self.model.wait_for_active_idle(self.wait_timeout, apps=apps),
         )
+
+    def _check_auto_restarts(self) -> None:
+        """Check if enable-auto-restarts is enabled.
+
+        If the enable-auto-restart is not enabled, this check will raise an exception.
+
+        :raises HaltUpgradePlanGeneration: When enable-auto-restarts is not enabled.
+        """
+        if "enable-auto-restarts" not in self.config:
+            logger.debug(
+                "%s application does not have an enable-auto-restarts config option", self.name
+            )
+            return
+
+        if self.config["enable-auto-restarts"].get("value") is False:
+            raise HaltUpgradePlanGeneration(
```

Comment:
IIRC, `HaltUpgradePlanGeneration` was created to be able to stop the plan generation of an app, but not stop the generation of other apps. Is this the intention here? I'm not sure if we should continue the plan generation in this case

Reply:
You are right, I need to change it to ApplicationError instead.